docs/algorithm-registry.md

```diff
@@ -0,0 +1,17 @@
+# Algorithm Registry
+
+This file is designed to act as a source of truth regarding what signing
+algorithms are permitted across the Sigstore ecosystem. Any changes to this file
+**must** be reflected in the `SupportedAlgorithm` enumeration in
+[sigstore_common.proto](../protos/sigstore_common.proto).
+
+Refer to the [Sigstore: Configurable Crypto Algorithms](https://docs.google.com/document/d/18vTKFvTQdRt3OGz6Qd1xf04o-hugRYSup-1EAOWn7MQ/)
```

Comment:
I wasn't sure how much detail to add here so I kept it short and just linked to the specification. If we want to flesh this out more in this doc instead of relying on the Google doc spec, I can do that.